src/app/pages/quickMaps/pages/baselineDetails/monthlyCard/monthlyCard.component.ts

```diff
@@ -0,0 +1,264 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
+import { MatTableDataSource } from '@angular/material/table';
+import { MonthlyFoodGroup } from 'src/app/apiAndObjects/objects/monthlyFoodGroup';
+import { MonthlyFoodGroups } from 'src/app/apiAndObjects/objects/monthlyFoodGroups';
+import { CurrentDataService } from 'src/app/services/currentData.service';
+import { QuickMapsService } from '../../../quickMaps.service';
+@Component({
+  selector: 'app-monthly-card',
+  templateUrl: './monthlyCard.component.html',
+  styleUrls: ['./monthlyCard.component.scss'],
+})
+export class MonthlyCardComponent implements OnInit {
+
+  @ViewChild(MatSort) sort: MatSort;
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+
+  public rawData: MonthlyFoodGroups;
+  public dataSource: MatTableDataSource<MonthlyFoodGroup>;
+  public chartData;
+  public cerealData;
+
+  public displayedColumns = [
+    'month',
+    // 'unitPerc',
+    'vegetablesPerc',
+    'cerealGrainsPerc',
+    'dairyPerc',
+    'fatPerc',
+    'fruitPerc',
+    'meatPerc',
+    'tubersPerc',
+    'nutsPerc',
+    'miscPerc',
+    'supplyTotal',
+    // 'supplyUnit',
+  ];
+
+  constructor(
+    private currentDataService: CurrentDataService,
+    private quickMapsService: QuickMapsService,
+  ) {
+
+    this.quickMapsService.parameterChangedObs.subscribe(() => {
+      this.currentDataService
+        .getMonthlyFoodGroups(
+          this.quickMapsService.countryId,
+          [this.quickMapsService.micronutrientId],
+          this.quickMapsService.popGroupId,
+          this.quickMapsService.mndDataId,
+        )
+        .then((data: MonthlyFoodGroups) => {
+          this.rawData = data;
+        })
+        .catch((err) => console.error(err))
+        .finally(() => {
+          this.initialiseGraph();
+          this.initializeTable(this.rawData.all);
+        });
+    });
+  }
+
+  ngOnInit(): void { }
+
+  public initialiseGraph(): void {
+    this.chartData = {
+      type: 'bar',
+      data: {
+        labels: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'],
+        datasets: [
+          {
+            label: 'Cereal Grains',
+            data: [
+              this.rawData.jan.cerealGrainsPerc,
```

Comment:
This could be simplified by doing a map on the all attribute:
this.rawData.all.map(group => group.cerealGrainsPerc);